devito/operator.py

```diff
@@ -121,89 +122,56 @@ def __init__(self, expressions, **kwargs):
         # Track the DLE output, as it might be useful at execution time
         self._dle_state = dle_state
 
+        self.symbolic_data = [x for x in parameters if isinstance(x, SymbolicData)]
+        dims = [x for x in parameters if isinstance(x, Dimension)]
+
+        d_parents = [x.parent for x in dims if hasattr(x, 'parent')]
+        self.dims = list(set(dims + d_parents))
+        assert(all(isinstance(x, RuntimeArgProvider)
+                   for x in self.symbolic_data + self.dims))
+        parameters = list(set(parameters + d_parents))
+        print(parameters)
         # Finish instantiation
         super(OperatorBasic, self).__init__(self.name, nodes, 'int', parameters, ())
 
+    def _reset_args(self):
```

Comment:
what's the purpose of this? I can kinda imagine it, but it needs a proper docstring